src/R/Components/Impl/Plots/Implementation/View/DragSurface.cs

```diff
@@ -7,15 +7,25 @@
 
 namespace Microsoft.R.Components.Plots.Implementation.View {
     internal class DragSurface {
-        private Point _lastMousedownPosition;
+        private Point? _lastMousedownPosition;
 
         public void MouseDown(MouseEventArgs e) {
             _lastMousedownPosition = e.GetPosition(null);
         }
 
+        public void MouseMove(MouseEventArgs e) {
+            if (e.MouseDevice.LeftButton == MouseButtonState.Released) {
+                _lastMousedownPosition = null;
+            }
+        }
+
+        public void MouseLeave(MouseEventArgs e) {
+            _lastMousedownPosition = null;
```

Comment:
These event are generally not very reliable. You may or may not get mouse leave. At least in Win32 that was the case. Usually events get lost of mouse moves to another window. Ex: press down, drag quickly, release over another window. 'leave' and 'release' may never come. I remember I used to check on mouse enter and mouse move if button was still down and reset if it was not.

but this may or may not be happening these days.